### PR TITLE
[risk=no][no ticket]Rename confusing "values" in DataSet and DataSetRequest

### PR DESCRIPTION
**Pull request by jmthibault79**

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/DataSetController.java [hunk 1]
```diff
@@ -362,9 +356,14 @@ public DomainValuePair apply(DataSetValues dataSetValue) {
                 "Timeout while querying the CDR to pull preview information.");
           }
 
+          final List<String> requestValues =
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/DataSetController.java]: This is #2630 which will get merged first so this diff will go away

File: ui/src/app/pages/data/data-set/export-data-set-modal.tsx [hunk 2]
```diff
@@ -111,13 +111,13 @@ class ExportDataSetModal extends React.Component<
   async exportDataSet() {
     this.setState({loading: true});
     const {dataSet, workspaceNamespace, workspaceFirecloudName} = this.props;
-    const request = {
+    const request: DataSetRequest = {
```
Comment (jmthibault79) [inline ui/src/app/pages/data/data-set/export-data-set-modal.tsx]: Always set the type, where possible

**Comment by jmthibault79** [comment]
Note to self: rebase after #2630 and #2573 have merged since this will conflict with them

File: api/src/main/java/org/pmiops/workbench/api/DataSetController.java [hunk 3]
```diff
@@ -132,23 +132,17 @@
...
-    } else if (dataSetRequest.getValues() == null || dataSetRequest.getValues().size() == 0) {
+    } else if (dataSetRequest.getDomainValuePairs() == null
+        || dataSetRequest.getDomainValuePairs().isEmpty()) {
       throw new BadRequestException("Missing values");
     }
     Timestamp now = new Timestamp(clock.instant().toEpochMilli());
     workspaceService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
         workspaceNamespace, workspaceId, WorkspaceAccessLevel.WRITER);
     long wId = workspaceService.get(workspaceNamespace, workspaceId).getWorkspaceId();
     List<DataSetValues> dataSetValuesList =
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/DataSetController.java]: NOT updating `DataSetValues`.  That is a whole other can of worms.  Aiming for minimal in this PR.

File: api/src/main/java/org/pmiops/workbench/db/model/DataSetValues.java [hunk 4]
```diff
@@ -34,10 +40,6 @@ public Domain getDomainEnum() {
     return CommonStorageEnums.domainFromStorage(Short.parseShort(domainId));
   }
 
-  public void setDomainEnum(Domain domain) {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/model/DataSetValues.java]: replaced by new constructor

File: ui/src/app/pages/data/data-set/dataset-page.tsx [hunk 5]
```diff
@@ -247,7 +248,7 @@ interface State {
   selectedCohortIds: number[];
   selectedConceptSetIds: number[];
   selectedPreviewDomain: string;
-  selectedValues: DomainValuePair[];
+  selectedValuePairs: DomainValuePair[];
```
Comment (ericsong) [inline ui/src/app/pages/data/data-set/dataset-page.tsx]: curious why you went with just `ValuePairs` in the front end and `DomainValuePairs` in the API.

Comment (jmthibault79) [reply ui/src/app/pages/data/data-set/dataset-page.tsx]: dunno - so the variable names wouldn't be overly long?  That's probably not a great reason.  I'll update to `selectedDomainValuePairs`

Comment (ericsong) [reply ui/src/app/pages/data/data-set/dataset-page.tsx]: I think thats a fair reason if you want to keep it as is.

Comment (ericsong) [reply ui/src/app/pages/data/data-set/dataset-page.tsx]: I don't think so BUT I am also a strong proponent for our naming being consistent between our code and our UI. Our UI currently calls these "Values" so I think our code should reflect that.

I do think there is a better name out there but I'd like product/design input and buy in.

Comment (jmthibault79) [reply ui/src/app/pages/data/data-set/dataset-page.tsx]: Yeah, my thought was that was non-ideal but unsafe to fix quickly

**Pull request merged**